src/pudl/settings.py

```diff
@@ -243,41 +244,82 @@ class Eia860Settings(GenericDatasetSettings):
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    all_eia860m_year_months: list[str] = DataSource.from_id(
+        "eia860m"
+    ).working_partitions["year_months"]
+    eia860m_year_months: list[str] = [max(all_eia860m_year_months)]
 
-    @field_validator("eia860m")
+    @field_validator("eia860m_year_months")
     @classmethod
-    def check_eia860m_year_month(cls, eia860m: bool) -> bool:
-        """Check 860m date-year is exactly one year after most recent working 860 year.
+    def add_other_860m_years(cls, v, info: FieldValidationInfo) -> list[str]:
+        """Find extra years from EIA860m if applicable.
 
-        Args:
-            eia860m: True if 860m is requested.
+        There's a gap in reporting (after the new year but before the EIA 860 early
+        release data in June) when we rely on two years worth of EIA860m data instead
+        of just one. This function adds the last available month_year values for each
+        year of 860m data that is not yet available in 860.
 
-        Returns:
-            eia860m: True if 860m is requested.
-
-        Raises:
-            ValueError: the 860m date is within 860 working years.
         """
-        eia860m_year = pd.to_datetime(cls.eia860m_year_month).year
-        expected_year = max(cls.data_source.working_partitions["years"]) + 1
-        if eia860m and (eia860m_year != expected_year):
-            raise AssertionError(
-                f"Attempting to integrate an eia860m year "
-                f"({eia860m_year}) from {cls.eia860m_year_month} not immediately following"
-                f"the eia860 years: {cls.data_source.working_partitions['years']}. "
-                f"Consider switching eia860m parameter to False."
-            )
-        return eia860m
+        if info.data["eia860m"]:
+            all_eia860m_years = {
+                date.split("-")[0] for date in info.data["all_eia860m_year_months"]
+            }
+            # The years in 860m that are not in 860
+            extra_eia860m_years = {
+                year
+                for year in all_eia860m_years
+                if int(year) not in info.data["years"]
+            }
+            # The years already listed as variables in eia860m_year_months
+            years_in_v = {date.split("-")[0] for date in v}
+            # The max year_month values available in 860m for each year not
+            # covered by EIA860 (and not already listed in the eia860m_year_months
+            # variable)
+            extra_eia860m_year_months = [
+                max(
+                    date
+                    for date in info.data["all_eia860m_year_months"]
+                    if date.startswith(year)
+                )
+                for year in (extra_eia860m_years - years_in_v)
+            ]
+            return v + extra_eia860m_year_months
+        return v
```

Comment:
To avoid having multiple return statements, could you always return
```
v + extra_eia860m_year_months
```
by initializing `extra_eia860m_year_months` to be an empty list?